### ISSUE TITLE
Share scanning process not working in Windows and macOS builds

### ISSUE THREAD

**Issue by mathiascode** [nicotine-plus/nicotine-plus#2608]

**Comment by mathiascode** [comment]
Fixed in d6afaf536729da5fa7f73d931e85cc9f9099855f (for Windows at least, need confirmation for macOS)

**Comment by slook** [comment]
@Vexariboerl @omidantilong @Michael-Z-Freeman @rpmindroid Please can somebody be so kind as to test [the latest build in master on macOS](https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos) to verify that the application loads shares normally upon startup and on menu item Rescan Shares? Thank you.

**Comment by omidantilong** [comment]
@slook it seems to load and rescan shares as normal. There were some other bugs in that latest build but as loading shares, it seems to behave 👍

<img width="460" alt="Screenshot 2023-08-08 at 01 18 53" src="https://github.com/nicotine-plus/nicotine-plus/assets/1922098/a0d552f0-79b1-4996-a753-8b7d71c8865f">

**Comment by mathiascode** [comment]
> There were some other bugs in that latest build

What are the bugs? I'm trying to get the development builds in shape for a 3.3.0 release, so it would be a good opportunity to report them.

**Comment by omidantilong** [comment]
I've not been able to test too thoroughly, but there seems to be some issues with connecting on startup, and again with quitting the app.

1) On launch, it doesn't seem to connect automatically, and using the "connect" option in the menu does nothing. Because it's not connecting, the search bar remains disabled. It looks like it's not getting past the login step.

<img width="1562" alt="Screenshot 2023-08-08 at 10 08 24" src="https://github.com/nicotine-plus/nicotine-plus/assets/1922098/56cef909-67b2-46e3-b78e-7f3b632b9cae">

2) When trying to quit using the menu option or Cmd+Q, the log pane shows "Quitting..." but nothing happens. The app remains responsive but can only be killed using a force quit. However unlike the issues from last time, force quitting is pretty snappy, and there's no issues re-opening the app after a force quit.

<img width="474" alt="Screenshot 2023-08-08 at 10 03 50" src="https://github.com/nicotine-plus/nicotine-plus/assets/1922098/685e9d43-bacb-4727-a1b9-3f6114b53f71">

(As you know I have a history with network issues 😅 I've just tested not without a VPN connection and with Little Snitch disabled and it's the same result)

**Comment by omidantilong** [comment]
Interesting, I've just let it sit for about 5 mins and now it's managed to connect. So this could possibly be an issue on my side. I will try and investigate more after work later today. Now that's it managed to connect, it's looking like a really good build 👍

**Comment by rpmindroid** [comment]
Don't know if that helps, but since I installed the last build I've had quite a few private chats from users complaining I wasn't sharing anything, even though I did choose a folder to share and click apply or whatever. So far I thought I was doing something wrong but now it seems there is a bug.

**Comment by mathiascode** [comment]
> Don't know if that helps, but since I installed the last build I've had quite a few private chats from users complaining I wasn't sharing anything, even though I did choose a folder to share and click apply or whatever. So far I thought I was doing something wrong but now it seems there is a bug.

Does this happen on startup after you've already queued downloads previously, or later? Do your files show up when you browse yourself (Menu -> Browse Public Shares)?

**Comment by mathiascode** [comment]
@omidantilong Debug logging could be useful here: https://nicotine-plus.org/doc/DEVELOPING.html#debug-logging

**Comment by omidantilong** [comment]
I think the behaviour @rpmindroid described is what I'm seeing too, where the search bar is disabled until the client connects. I just tried again, opened Nicotine and left open and it did finally connect but it took at least 5 mins. VPN/Little Snitch/TripMode/etc doesn't seem to make any difference.

**Comment by rpmindroid** [comment]
> @Vexariboerl @omidantilong @Michael-Z-Freeman @rpmindroid Please can somebody be so kind as to test [the latest build in master on macOS](https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos) to verify that the application loads shares normally upon startup and on menu item Rescan Shares? Thank you.

Sorry, my very bad, I missed that one message and I replied too quick. What I called my ‘last build’, the one with the issues, was from a week or ten days ago. I just reinstalled N+ with the latest one and everything seems to be working fine for me (macOS): N+ connects / search field appears after about 6 seconds (much shorter than before), and I asked people in the room chat to check if they could browse my files and one person tried and succeeded. Also I can browse my own shares, and I downloaded a couple albums to see if the user on the other end would react, and nothing happened. So it looks like it's all good.
